Dockerfile

```diff
@@ -1,35 +1,70 @@
-FROM python:3.9.7-slim@sha256:cd1045dbabff11dab74379e25f7974aa7638bc5ad46755d67d0f1f1783aee101
+# syntax=docker/dockerfile:1.3
 
-ENV PYTHONUNBUFFERED=1 \
-    PYTHONPATH=/app/ \
-    PORT=8888
+FROM python:3.9-slim-bullseye as compile
 
-EXPOSE $PORT
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
+    # Needed to download Rust
+    curl \
+    # Needed to build psycopg and uWSGI 
+    build-essential \
+    python-dev \
+    libpq-dev \
+    # Needed to build uwsgi-dogstatsd plugin
+    git
 
-# add a non-privileged user for installing and running
-# the application
-RUN mkdir /app && \
-    chown 10001:10001 /app && \
-    groupadd --gid 10001 app && \
-    useradd --no-create-home --uid 10001 --gid 10001 --home-dir /app app
+# Get rustup https://rustup.rs/ for canonicaljson-rs
+# minimal profile https://rust-lang.github.io/rustup/concepts/profiles.html
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --profile minimal -y
+# Add cargo to PATH
+RUN . ~/.cargo/env
+
+RUN python -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
 
 COPY requirements.txt .
-COPY bin/docker-install.sh .
-RUN ./docker-install.sh
 
-# TODO: don't copy any /tests folders for prod image (maybe handle this in .dockerignore)
-COPY . /app
+# Python packages
+RUN pip install --upgrade pip
+RUN apt-get install -y --no-install-recommends libpq-dev
+RUN pip install --no-cache-dir -r requirements.txt
```

Comment:
Could shrink down to one `RUN`

Reply:
Good call on combining the other RUN commands. For this one though, I'm going to:
```
...

RUN pip install --upgrade pip
COPY requirements.txt .
RUN pip install --no-cache-dir -r requirements.txt
```

Since I think this will be a better strategy from a caching perspective.

Also, we can be a little less careful with image size for this intermediate stage since we only end up copying the `venv` and UWSGI plugin from here to the final image. Combining `RUN` commands will probably be more important there.

Reply:
Also, I don't know why this line
```
RUN apt-get install -y --no-install-recommends libpq-dev
```

is here since we already install dependencies from `apt` above, so I'm going to delete this line.

Reply:
Makes sense! 👍🏻

Reply:
https://github.com/mozilla-services/kinto-dist/pull/1937/commits/f7de5cb0ece8da0fd2c94dc8351bb41363155028https://github.com/mozilla-services/kinto-dist/pull/1937/commits/f7de5cb0ece8da0fd2c94dc8351bb41363155028